### DISCUSSION TITLE
Porous Flow permeability equivalent of GenericFunctionRankTwoTensor

### DISCUSSION THREAD

**Discussion by ln53** [idaholab/moose#23699] [Q&A Modules: General] [answered]
Hi,
Is there an easy way to use GenericFunctionRankTwoTensor to define the permeability tensor in the Porous Flow module? I've tried modifying PorousFlowPermeabilityConst to accept the name of a GenericFunctionRankTwoTensor (the functions are constant in time), but it hasn't been as straightforward as I hoped. I've also looked at the other PorousFlowPermeability... options and none of them seem to do this.
Thanks.

**Comment by GiudGiud** [comment, marked as answer]
I had a quick look and this should be doable. 

Are you planning on using an input material property or an several functions, one for each component?

**Reply by ln53** [reply]
Thanks - I will be using a function for each component. I have already plugged them in to a GenericFunctionRankTwoTensor though, so would be happy to just supply that as an input material property if easier.

**Reply by GiudGiud** [reply]
Sounds good. 

I think you could make a PorousFlowPermeabilityFromMatProp object

Please let me know where you have issues making it and I can advise you

**Reply by GiudGiud** [reply]
another option would be to name your already existing material property with the expected name for the permeability.

have you tried this?

**Reply by cpgr** [reply]
> Sounds good.
> 
> I think you could make a PorousFlowPermeabilityFromMatProp object
> 
> Please let me know where you have issues making it and I can advise you

I think this would be the best idea!

**Reply by ln53** [reply]
Thanks for the input - after another attempt I did manage to create a "PorousFlowPermeabilityFromMatProp" object as suggested, and it's working as I'd hoped! I've attached the source and header files here (in txt format) in case they're useful to anyone in future.
[PorousFlowPermeabilityFromMatPropC.txt](https://github.com/idaholab/moose/files/10968560/PorousFlowPermeabilityFromMatPropC.txt)
[PorousFlowPermeabilityFromMatProph.txt](https://github.com/idaholab/moose/files/10968561/PorousFlowPermeabilityFromMatProph.txt)

**Reply by GiudGiud** [reply]
@cpgr are you interested in the contribution of these to the module?

**Reply by cpgr** [reply]
If @ln53 is happy to contribute this I'm sure others might find it useful

**Reply by ln53** [reply]
Will look into it 👍
